### PR TITLE
Handle JSON RPC errors. Resolves issue #672.

### PR DESCRIPTION
**Pull request by rossmcf**

### CODE REVIEW

File: transport/http/jsonrpc/encode_decode.go [hunk 1]
```diff
@@ -40,9 +40,9 @@ type EncodeResponseFunc func(context.Context, interface{}) (response json.RawMes
 // JSON encodes the object directly.
 type EncodeRequestFunc func(context.Context, interface{}) (request json.RawMessage, err error)
 
-// DecodeResponseFunc extracts a user-domain response object from an HTTP
-// request object. It's designed to be used in JSON RPC clients, for
+// DecodeResponseFunc extracts a user-domain response object from an JSONRPC
+// response object. It's designed to be used in JSON RPC clients, for
 // client-side endpoints. One straightforward DecodeRequestFunc could be
```
Comment (peterbourgon) [inline transport/http/jsonrpc/encode_decode.go]: Oops, looks like a copy/paste flub in the comment! Probably there are others of these, too. Also, does the actual comment from this point on make sense?

Comment (rossmcf) [reply transport/http/jsonrpc/encode_decode.go]: Thanks for spotting that. I've made some tweaks to the doc comments in that file.

**Comment by rossmcf** [comment]
I think this looks tidy now. I'll be using it in anger in the near future, so will feed back if there are any other tweaks needed.

**Comment by basvanbeek** [comment]
@peterbourgon before merge I'd like to play a bit with it... can you wait max. a day until merge?

**Comment by peterbourgon** [comment]
@basvanbeek I'll wait until you ping me again 👍

**Comment by peterbourgon** [comment]
Failures are related to the Thrift nonsense, so I'm gonna optimistically merge.

**Pull request merged**